crates/proof-of-sql/src/base/math/decimal.rs

```diff
@@ -42,13 +31,17 @@ pub enum DecimalError {
 
     /// Errors that may occur when parsing an intermediate decimal
     /// into a posql decimal
-    #[error("Intermediate decimal conversion error: {0}")]
-    IntermediateDecimalConversionError(IntermediateDecimalError),
+    #[error(transparent)]
+    IntermediateDecimalConversionError(#[from] IntermediateDecimalError),
 }
 
-impl From<IntermediateDecimalError> for ConversionError {
-    fn from(err: IntermediateDecimalError) -> ConversionError {
-        DecimalConversionError(IntermediateDecimalConversionError(err))
+/// Result type for decimal operations.
+pub type DecimalResult<T> = Result<T, DecimalError>;
+
+// This exists because `TryFrom<arrow::datatypes::DataType>` for `ColumnType` error is String
+impl From<DecimalError> for String {
+    fn from(error: DecimalError) -> Self {
+        error.to_string()
```

Comment:
Since we're here right now, it might be prudent to actually go implement the error variant `ColumnType` error